### PR TITLE
CTO-120 Updated endpoint and controller for gate changes

### PR DESCRIPTION
**Pull request by zack-rma**
Added operational change store/delete support

### CODE REVIEW

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeEndpointInput.java [hunk 1]
```diff
@@ -105,11 +103,11 @@ public static final class Post extends EndpointInput {
         static final boolean DEFAULT_FAIL_IF_EXISTS = true;
         static final String FAIL_IF_EXISTS = "fail-if-exists";
         private boolean failIfExists = true;
-        private final Set<GateChange> changes = new HashSet<>();
+        private final Set<GateChange> changes;
 
         private Post(Set<GateChange> changes) {
             super();
-            changes.addAll(Objects.requireNonNull(changes, "Set of changes is required to be non-null"));
+            this.changes = Objects.requireNonNull(changes, "Set of changes is required to be non-null");
```
Comment (RyanM-RMA) [inline cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeEndpointInput.java:110]: Why did you change this?

Comment (zack-rma) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeEndpointInput.java:110]: Changes were not being added to the set correctly. The result of attempting to add values was an empty set.

Comment (zack-rma) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeEndpointInput.java]: Fixed in latest push

**Review by RyanM-RMA — DISMISSED** [review]

File: cwms-radar-model/src/main/java/mil/army/usace/hec/cwms/radar/client/model/GateChange.java [hunk 2]
```diff
@@ -40,6 +44,9 @@ public class GateChange {
     @JsonProperty("notes")
     private String notes = null;
 
+    @JsonProperty("type")
```
Comment (rma-bryson) [inline cwms-radar-model/src/main/java/mil/army/usace/hec/cwms/radar/client/model/GateChange.java:47]: Is this a default type for this? But it can be changed?

Should GetSetting also have similar behavior with a default type?

Comment (zack-rma) [reply cwms-radar-model/src/main/java/mil/army/usace/hec/cwms/radar/client/model/GateChange.java:73]: This is part of the sub-typing setup for serialization/deserialization. I was able to remove the getter/setter without losing functionality.

I also added a default to GateSetting for consistency.

**Review by rma-bryson — APPROVED** [review]
